src/indexing.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    CheckpointOutput
+
+This is a index entry describing the output file from a checkpoint.
+You can retrieve a list of these from a folder full of such outputs, using
+[`index_checkpoint_files`](@ref).
+
+For accessing details of the CheckpointOutput the following helpers are provided:
+[`path`](@ref), [`name`](@ref), [`groups`](@ref), [`tags`](@ref).
+Further: `getproperty` is overloaded so that you can access the value of the tag `:foo` via
+`x.foo`.
+"""
+struct CheckpointOutput{P<:AbstractPath}
+    path::P
+    name::AbstractString
+    groups::NTuple{<:Any, AbstractString}
+    tags::NTuple{<:Any, Pair{Symbol, <:AbstractString}}
+end
+
+path(x::CheckpointOutput) = getfield(x, :path)
+name(x::CheckpointOutput) = getfield(x, :name)
+groups(x::CheckpointOutput) = getfield(x, :groups)
+tags(x::CheckpointOutput) = getfield(x, :tags)
+_tag_names(x::CheckpointOutput) = first.(tags(x))
+
+#Tables.columnnames(x::CheckpointOutput) = propertynames(x)
+Base.propertynames(x::CheckpointOutput) =  [:groups, :name, _tag_names(x)..., :path]
+
+function Base.getproperty(x::CheckpointOutput, name::Symbol)
+    inds = findall(==(name)∘first, tags(x))
+    if length(inds) > 1
+        error(
+            "The checkpoint was tagged with $name multiple times (positions $inds). " *
+            "you thus can not use the getproperty shorthand to check its tag, as we don't" *
+            "know which you mean. Use the `tags` function instead."
+        )
+    elseif length(inds) == 0
+        if hasfield(CheckpointOutput, name)
+            # as long as we don't have a tag with one of the field names we pass it through
```

Comment:
I am not sure about this. Sure, the fields are not public API, but changing the behaviour given which tags are used seems evil. How about we error here and tell the user to use the methods?

Reply:
we could.
Main reason I don't want to is so that tables work.
But we can make tables fail in that case also.

Reply:
Ok, I leave it up to you to decide. LGTM overall

Reply:
I have made that change